### DISCUSSION TITLE
Goreleaser NFPMS build Issue

### DISCUSSION THREAD

**Discussion by rupamkhaitan** [goreleaser/goreleaser#1627] [Q&A] [answered]
@caarlos0 i am using nfpms: to generate debian build for go package and i am able to customize the debian name when generating but the control file is appending -SNAPSHOT-<random_hash>

While installing debian on the linux the version field comes with -SNAPSHOT-<random_hash>

how can i fix the control file of debian to not append this SNAPSHOT

![image](https://user-images.githubusercontent.com/3277629/85615218-859c2400-b621-11ea-8d49-bb02f0f71ab1.png)

**Comment by caarlos0** [comment, marked as answer]
its not a random hash, is the commit sha.

you're running with `--snapshot`, on production that would not happen... 

anyway you can customize the snapshot version: https://goreleaser.com/customization/snapshots/

**Comment by rupamkhaitan** [comment]
so you mean running this way
```
goreleaser release --snapshot --rm-dist
```

**Reply by rupamkhaitan** [reply]
And i changed the snapshot name template to use just Tag (which we tag for our versions)

name_template: "{{ .Tag }}"

**Reply by rupamkhaitan** [reply]
I got it.. i need to make  release in the template

**Reply by caarlos0** [reply]
if you run against a tag, you don't need the `--snapshot`

**Comment by rupamkhaitan** [comment]
@caarlos0 is there any way to append exact version to be displayed in the dpkg -s when doing release build?

Say my tag is **1.0.0-b1-noc** and i made a release build of debian and install in the linux (say my debian pkg name is **go-broker**)
When i install it

```
dpkg -i go-broker
```

Later i run to check if it installed and whats the version
```
dpkg -s go-broker
```

It appends in different format..

![image](https://user-images.githubusercontent.com/3277629/85635955-cb1e1880-b644-11ea-8f67-bcce1bd734f2.png)

Is it possible to show exact version like 1.0.0-b1-noc

**Reply by rupamkhaitan** [reply]
@caarlos0  OK I got the reason why it was adding this additional 1, i commented out
```
# epoch: 1
# release: 1
```
So all the additional -1 went off.. now only thing is not sure why it adds tilde(~) before b1-noc

New version display after commenting out epoch & release property

![image](https://user-images.githubusercontent.com/3277629/85636345-ea697580-b645-11ea-8e75-07746524c148.png)
